### DISCUSSION TITLE
Toolbar with input that acts as a select

### DISCUSSION THREAD

**Discussion by bengry** [ariakit/ariakit#4239] [Questions] [answered]
I'm trying to achieve a component that's a composition of three components - Toolbar, Select, and an input, but can't figure out how to make everything tick together.

The gist is that the toolbar should have an `<input />` that lets the user either ender a date manually, but also - when it's focused, a popover opens with a list of preset options to choose from, for easier usage.

I have two issues right now:
1. When the Select popover opens - the input looses focus, even though `virtualFocus={true}` (on `<SelectProvider>`). I also tried adding `autoFocusOnShow={false}` on the `<SelectPopover>`, but to no avail.
2. I can't get the focus to work quite right - `<Select render={<ToolbarItem />}` work fine, but when adding an `<input>` into the mix things fail. Right now I'm rendering the `<input>` _inside_ the composed `Select` + `ToolbarItem`, but that's not what I want exactly, since it missed out on the focus from the toolbar.

Here's a quick repro: https://stackblitz.com/edit/qqtrzo?file=toolbar-select%2Findex.tsx

Any help would be appreciated.

Thanks!

P.S.
Is this select considered a Combox? Or maybe even a [Dropdown](https://ariakit.org/components/menu) I'm not 100% sure here.

**Comment by diegohaz** [comment, marked as answer]
An `input that acts as a select` is essentially a [Combobox](https://ariakit.org/components/combobox).

Here's an example: https://stackblitz.com/edit/qqtrzo-vrebtc?file=toolbar-select%2Findex.tsx

**Reply by bengry** [reply]
Thanks! I managed to get this working with this example. Much appreciated.
